service/service.go

```diff
@@ -61,19 +58,11 @@ type Service struct {
 
 // New creates a new service with given configuration.
 func New(config Config) (*Service, error) {
-	if config.Logger == nil {
```

Comment:
I still think checking dependencies only down the chain is unmanageable.

Reply:
I think the opposite is true. What exactly is unmanageable in your opinion?

Reply:
Let's say at some point someone starts to use logger in this Service. I doubt he will remember to add a check here as well.